Source/Pawnmorphs/Esoteria/Hediffs/Base/Hediff_PM.cs

```diff
@@ -0,0 +1,543 @@
+        /// </summary>
+        public override void PostAdd(DamageInfo? dInfo)
+        {
+            base.PostAdd(dInfo);
+            foreach (HediffComp_PM pmComp in PMComps)
+                pmComp.CompPostPostAdd(dInfo);
+            PostAddOrLoad();
+        }
+
+        /// <summary>
+        /// Called after the hediff is first added to the pawn, and any time the game is loaded (during the
+        /// <see cref="LoadSaveMode.PostLoadInit" /> phase).
+        /// <br />
+        /// Any initialization that needs to be repeated after every load can be done here.
+        /// </summary>
+        protected virtual void PostAddOrLoad()
+        {
+            HashOffset = pawn?.HashOffset() ?? 0;
+        }
+
+        /// <summary>
+        /// Called after the hediff is removed from the pawn.
+        /// </summary>
+        public override void PostRemoved()
+        {
+            base.PostRemoved();
+            foreach (HediffComp_PM pmComp in PMComps)
+                pmComp.CompPostPostRemoved();
+        }
+
+        //TODO - override Tick() the same way Hediff_StageChanges does
+
+        /// <summary>
+        /// Called after the main Hediff tick.  Comps and PMComps are ticked in this method.
+        /// <br />
+        /// Note that this method is a MAJOR hot path -- it should be optimized as much as possible because it potentially gets
+        /// called tens of thousands of times per second. For this reason, foreach loops should be avoided because the iterator
+        /// object creation is a notable source of slowdown.
+        /// </summary>
+        public override void PostTick()
+        {
+            // We're not calling base.PostTick() here because we're manually replicating it in a slightly more optimized way
+            var severityAdjustment = 0f;
+
+            if (_hasComps && comps != null)
+            {
+                int compCount = comps.Count; // Not caching this genuinely has a noticeable performance impact for hediffs that
+                // have multiple comps.  We can't cache it beyond the tick, however, because
+                // exceptions in certain places can cause comps to be removed.
+                // ReSharper disable once ForCanBeConvertedToForeach
+                for (var index = 0; index < compCount; ++index)
+                    comps[index]?.CompPostTick(ref severityAdjustment);
+            }
+
+            if (_hasPMComps)
+            {
+                int compCount = _pmComps.Count; // Same as above
+                int hashOffsetTick = Find.TickManager!.TicksGame + HashOffset;
+
+                // Note that PMComps.CompPostTick() is never called.  This is because that method exists only for legacy support.
+                // That method cannot be overridden and only does exactly what we're doing here (but slightly less efficiently)
+
+                if (hashOffsetTick % 60 == 0) // Every real-life second
```

Comment:
The interesting part here to consider is that this is only a real-life second at 1x speed with no lag.
But we could potentially increase delay on higher gameplay speeds and do more interpolation to compensate if we wanted to.
If we go by "Run once every RL tick regardless of game speed and provide a count of how many ticks has past since last call."

Reply:
Honestly, I prefer that it be synced to game speed instead of real-life speed.  Otherwise we get the possibility of non-deterministic behavior that depends on how fast the game is runs on your machine, which is just a headache.

I can change the comment to clarify that that's only at 1x speed without lag, though.